app/screens/sso/sso_with_webview.tsx

```diff
@@ -0,0 +1,232 @@
+        let isMessagingEnabled = false;
+        const parsed = urlParse(url);
+        if (parsed.host.includes('.onelogin.com')) {
+            setJSCode(oneLoginFormScalingJS);
+        } else if (parsed.pathname === completeUrlPath) {
+            // To avoid `window.postMessage` conflicts in any of the SSO flows
+            // we enable the onMessage handler only When the webView navigates to the final SSO URL.
+            isMessagingEnabled = true;
+        }
+        setMessagingEnabled(isMessagingEnabled);
+    };
+
+    const onLoadEnd = (event: WebViewNavigationEvent | WebViewErrorEvent) => {
+        const url = event.nativeEvent.url;
+        const parsed = urlParse(url);
+
+        let isLastRedirect = url.includes(completeUrlPath);
+        if (ssoType === ViewTypes.SAML) {
+            isLastRedirect = isLastRedirect && !parsed.query;
+        }
+
+        if (isLastRedirect) {
+            extractCookie(parsed);
+        }
+    };
+
+    const renderWebView = () => {
+        if (shouldRenderWebView) {
+            return (
+                <WebView
+                    automaticallyAdjustContentInsets={false}
+                    cacheEnabled={false}
+                    injectedJavaScript={jsCode}
+                    javaScriptEnabled={true}
+                    onLoadEnd={onLoadEnd}
+                    onMessage={messagingEnabled ? onMessage : undefined}
+                    onNavigationStateChange={onNavigationStateChange}
+                    onShouldStartLoadWithRequest={() => true}
+                    ref={webView}
+                    source={{uri: loginUrl, headers: HEADERS}}
+                    startInLoadingState={true}
+                    userAgent='Mozilla/5.0 (Linux; Android 10; Android SDK built for x86 Build/LMY48X) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/81.0.4044.117 Mobile Safari/608.2.11'
+                    useSharedProcessPool={false}
+                />
+            );
+        }
+        return <Loading/>;
+    };
+
+    return (
+        <SafeAreaView
+            style={style.container}
+            testID='sso.webview'
+        >
+            <StatusBar/>
+            {error || loginError ? (
+                <View style={style.errorContainer}>
+                    <Text style={style.errorText}>{error || loginError}</Text>
+                </View>
+            ) : renderWebView()}
+        </SafeAreaView>
+    );
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flex: 1,
+        },
+        errorContainer: {
+            alignItems: 'center',
+            flex: 1,
+            marginTop: 40,
+        },
+        errorText: {
+            color: changeOpacity(theme.centerChannelColor, 0.4),
+            fontSize: 16,
+            fontWeight: '400',
+            lineHeight: 23,
+            paddingHorizontal: 30,
+        },
+    };
+});
+
+export default React.memo(SSOWithWebView);
```

Comment:
Any reason why this is being memoized ?

Reply:
It's not needed here .. just a habit of doing it by default to prevent not-needed re-renders. Removing it 👍